qa/tasks/ceph_test_case.py

```diff
@@ -134,9 +138,12 @@ def __exit__(self, exc_type, exc_val, exc_tb):
                 except CommandFailedError:
                     pass
 
-                if not self.match():
+                if emit and not self.match():
                     log.error("Log output: \n{0}\n".format(self.watcher_process.stdout.getvalue()))
                     raise AssertionError("Expected log message not found: '{0}'".format(expected_pattern))
+                elif fail or (not emit and self.match()):
```

Comment:
Isn't `not emit and self.match()` redundant here? `fail` is `True` if `not emit and self.match()` in line 127. This could just be `elif fail`?

Reply:
No, these calls to `self.match` also check if there were any logged messages after `time.sleep(5 + timeout)`.

Reply:
ah, got it. makes sense